### PR TITLE
fix(dynamic-config): skip keys that are used in the configuration file

### PR DESCRIPTION
**Pull request by JoergSiebahn**

### CODE REVIEW

File: sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/LookupTracker.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.sdase.commons.server.dropwizard.bundles;
+
+import java.util.Set;
+
+/** Tracks keys that where looked up in the environment. */
```
Comment (christopher-cudennec) [inline sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/LookupTracker.java:5]: nit: "that **were** looked up"

File: sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/configuration/generic/GenericLookupYamlConfigurationFactory.java [hunk 2]
```diff
@@ -72,6 +78,7 @@ protected T build(JsonNode node, String path) throws IOException, ConfigurationE
         .map(m -> m.expand(configurationRuntimeContext.getDefinedKeys()))
         .flatMap(List::stream)
         .sorted()
+        .filter(m -> !lookupTracker.lookedUpKeys().contains(m.getContextKey()))
```
Comment (christopher-cudennec) [inline sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/configuration/generic/GenericLookupYamlConfigurationFactory.java:81]: `lookedUpKeys()` will create a new unmodifiable `Set` for every invocation (see `SystemPropertyAndEnvironmentLookup`). The `LookupTracker` could expose a `contains` method that returns `true` or `false`.

File: sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/LookupTracker.java (hunk 1 above)
Comment (JoergSiebahn) [reply sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/LookupTracker.java]: fixed … guess that's my most common mistake in writing

File: sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/configuration/generic/GenericLookupYamlConfigurationFactory.java (hunk 2 above)
Comment (JoergSiebahn) [reply sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/configuration/generic/GenericLookupYamlConfigurationFactory.java]: If you are afraid of the performance, what about [this solution](https://github.com/SDA-SE/sda-dropwizard-commons/compare/0e35106bf841353b38670b3a87f2b6f68bcd2c75..9946c82cd462e5d78c328ceacf713038779914b5)?

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**